HHS/simpler-grants-gov
# [Issue #1563] Ability to hide search filters in mobile viewport

## Summary
Fixes #1563

Todos:
- [ ] refactor to reduce duplication of search page on error page (see https://github.com/HHS/simpler-grants-gov/issues/2260)

### Time to review: __60 mins__

## Changes proposed

Adds logic for hiding / showing filter options on search page such that:

* on viewports larger than mobile
    * no change
* on mobile viewports
    * all filter options are hidden by default
    * a button appears to allow toggling of appearance of filters
    * toggle button icon changes from up to down arrow when toggled

### Technical changes

* creates ContentDisplayToggle to handle toggle functionality
* creates SearchFilters component to isolate client side state control
* refactors to search page and additional styling to allow for this change
* supporting tests and story

Also:

* adds clsx package
* updates eslint rules for no-unused-args to allow for underscore prefixed args to be unused


## Context for reviewers

To make this change the way that seemed best for me, I made a number of related but perhaps not entirely necessary changes. I'm happy to discuss any of these change further and take a different approach

### Test steps

1. visit your local search page with desktop width viewport
2. _VALIDATE_: all filter options (accordions and checkboxes) appear as normal
3. shrink to mobile size viewport (ex 375px)
4. _VALIDATE_: all filter options are hidden
5. _VALIDATE_: filter toggle button appears matching designs from ticket
6. click filter toggle button
7. _VALIDATE_: filters appear
8. click filter toggle button
7. _VALIDATE_: filters disappear
8. resize to desktop viewport size
7. _VALIDATE_: filters appear 

## Additional information


https://github.com/user-attachments/assets/0fba089b-2057-4cf5-87b9-0ae410f5b3d2
